README.md

```diff
@@ -0,0 +1,140 @@
+contains a runner and language-specific harnesses to confirm feature behavior across versions.
+
+These SDK features serve several purposes:
+
+* Ensure parity across SDKs by having same-feature snippets adjacent to one another
+* Confirm feature behavior across SDK versions
+* Confirm history across SDK versions
+* Document features in different SDKs
+* Easy-to-use environment for writing quick workflows in all languages/versions
+
+## Building
+
+With latest [Go](https://golang.org/) installed, run:
+
+    go build
+
+## Running
+
+Prerequisites:
+
+* [Go](https://golang.org/) 1.17+
+* [JDK](https://adoptium.net/?variant=openjdk11&jvmVariant=hotspot) 11+
+* [Node](https://nodejs.org) 16+
+
+Command:
+
+    sdk-features run --lang LANG [--version VERSION] [PATTERN...]
+
+Note, `go run .` can be used in place of `go build` + `sdk-features` to save on the build step.
+
+`LANG` can be `go`, `java`, or `ts`. `VERSION` is per SDK and if left off, uses the latest version set for the language
+in this repository.
+
+`PATTERN` must match either the features relative directory _or_ the relative directory + `/feature.<ext>` via
+[Go path match rules](https://pkg.go.dev/path#Match) which notably does not include recursive depth matching. If
+`PATTERN` arguments are not present, the default is to run all features.
+
+Several other options are available, some of which are described below. Run `sdk-features run --help` to see all
+options.
+
+**NOTE** There is [currently a bug](https://github.com/temporalio/temporal/issues/2207) with Temporalite that causes
+history checks to fail. Until it is fixed, the external server as shown below must be used.
+
+### External Server and Namespace
+
+By default, a [temporalite](https://github.com/DataDog/temporalite) is dynamically started at runtime to handle all
+feature runs. To not start the embedded server, use `--server` to specify the address of a server to use. Similarly, to
+not use a custom namespace that may not be registered on the external server, use `--namespace`
+
+### History Checking
+
+History files are present at `features/<path/to/feature>/history/history.<lang>.<version>.json`. By default, three
+history checks are performed:
+
+1. Specific languages use their replayers to replay the just-executed workflow's history to confirm it works.
+2. Specific languages use their replayers to replay all history files with versions <= the current version to confirm it
+   works.
+3. The primary runner scrubs the history of the just-executed workflow of all execution-dependent values. Then it
+   compares the exact events to all similarly-scrubbed history files.
+
+Currently there are not ways for features to opt out of specific history checks. To opt out of all history checking for
+a specific run, use `--no-history-check`.
+
+## Writing Features
+
+Developers can write workflows and activities that represent a SDK "feature". These are organized into directories under
+the `features/` directory.
```

Comment:
We should consider a dir for things that aren't really features but we'd like to include in the "spec". No need to change this now but just FYI

Reply:
I am doing that with `features/bugs`. But maybe it just doesn't read properly and I should just call this top level "spec" instead of "features"?

Reply:
I think that makes good sense honestly. Each sub-folder under spec will have it's little readme/bit of info about what it is anyway.

Reply:
I have opened https://github.com/temporalio/sdk-features/issues/4 to address this after these PRs are merged. Doing this across PRs is quite annoying.